agent/agent_test.go

```diff
@@ -126,6 +126,24 @@ func (s *AgentSuite) TestAgentEndTaskShouldExit() {
 	s.Error(err)
 }
 
+func (s *AgentSuite) TestNextTaskConflict() {
+	s.mockCommunicator.NextTaskShouldConflict = true
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
time this out?

Reply:
No need. There is a default case in the select, which is what we want to happen.